### PR TITLE
Handling case when Jwk doesn't have 'use' parameter

### PR DESCRIPTION
**Pull request by manpreet-compro**
#113 
# Description

Updated the code to handle the case when 'use' parameter in jwk key is not present on is not having any value. As per spec, https://tools.ietf.org/html/rfc7517#section-4.2 
> Use of the "use" member is OPTIONAL, unless the application  requires its presence.

Added a check to check for presence first, if key is present and having non blank value then compare if value is 'sig' as per spec. 

### References

This PR is submitted in response to Open issue #24 

### Testing
Updated test cases to include 2 test cases
-  value of 'use' is blank (Should pass)
-  value of 'use' is 'enc' (Should fail)

Command `npm run test` can be used to test the implementation

### Checklist

- [x] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `master`

### CODE REVIEW

File: src/JwksClient.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ export class JwksClient {
       }
 
       const signingKeys = keys
-        .filter(key => key.use === 'sig' && key.kty === 'RSA' && key.kid && ((key.x5c && key.x5c.length) || (key.n && key.e)))
+        .filter(key => (!key.use || key.use === 'sig') && key.kty === 'RSA' && key.kid && ((key.x5c && key.x5c.length) || (key.n && key.e)))
```
Comment (lbalmaceda) [inline src/JwksClient.js]: Is there a way we can improve this filtering condition? Each clause we add makes it longer and harder to read.

File: tests/jwksClient.tests.js [hunk 2]
```diff
@@ -278,7 +278,25 @@ describe("JwksClient", () => {
...
+              kid: "IdTokenSigningKeyContainer.v3",
+              nbf: 1459289287,
+              use: "",
+              kty: "RSA",
+              e: "AQAB",
+              n:
+                "s4W7xjkQZP3OwG7PfRgcYKn8eRYXHiz1iK503fS-K2FZo-Ublwwa2xFZWpsUU_jtoVCwIkaqZuo6xoKtlMYXXvfVHGuKBHEBVn8b8x_57BQWz1d0KdrNXxuMvtFe6RzMqiMqzqZrzae4UqVCkYqcR9gQx66Ehq7hPmCxJCkg7ajo7fu6E7dPd34KH2HSYRsaaEA_BcKTeb9H1XE_qEKjog68wUU9Ekfl3FBIRN-1Ah_BoktGFoXyi_jt0-L0-gKcL1BLmUlGzMusvRbjI_0-qj-mc0utGdRjY-xIN2yBj8vl4DODO-wMwfp-cqZbCd9TENyHaTb8iA27s-73L3ExOQ"
+            },
+            {
+              kid: "IdTokenSigningKeyContainer.v3",
```
[the remark this answers is not in the document]
Comment (manpreet-compro) [reply tests/jwksClient.tests.js]: Agree. Updated

File: tests/jwksClient.tests.js [hunk 3]
```diff
@@ -278,7 +278,25 @@ describe("JwksClient", () => {
               e: "AQAB",
               n:
                 "s4W7xjkQZP3OwG7PfRgcYKn8eRYXHiz1iK503fS-K2FZo-Ublwwa2xFZWpsUU_jtoVCwIkaqZuo6xoKtlMYXXvfVHGuKBHEBVn8b8x_57BQWz1d0KdrNXxuMvtFe6RzMqiMqzqZrzae4UqVCkYqcR9gQx66Ehq7hPmCxJCkg7ajo7fu6E7dPd34KH2HSYRsaaEA_BcKTeb9H1XE_qEKjog68wUU9Ekfl3FBIRN-1Ah_BoktGFoXyi_jt0-L0-gKcL1BLmUlGzMusvRbjI_0-qj-mc0utGdRjY-xIN2yBj8vl4DODO-wMwfp-cqZbCd9TENyHaTb8iA27s-73L3ExOQ"
-            }
+            },
+            {
+              kid: "IdTokenSigningKeyContainer.v3",
+              nbf: 1459289287,
+              use: "",
```
[the remark this answers is not in the document]
Comment (manpreet-compro) [reply tests/jwksClient.tests.js]: Yes,  "optional" in spec implies the key's presence is optional and is not required to be present to be always. I have updated test case to remove this key. 

However this brings another question, Currently I have added check of `!key.use` which will pass for all javascript [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) values. Should we continue with that or add strict check of key presence and enforce that if key is present then it must have value equal to 'sig'. Your thoughts ?

File: src/JwksClient.js (hunk 1 above)
Comment (manpreet-compro) [reply src/JwksClient.js]: Please refer comment on other thread, https://github.com/auth0/node-jwks-rsa/pull/116#discussion_r365780322

Once we finalize that, I will try to improve the checks to make it more readable

File: tests/jwksClient.tests.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (lbalmaceda) [reply tests/jwksClient.tests.js]: @manpreet-compro What about `hasOwnProperty` ?

File: src/JwksClient.js (hunk 1 above)
Comment (manpreet-compro) [reply src/JwksClient.js]: One possible way to simplify filter function is following, i.e. break && conditions into multiple if statements

```
const signingKeys = keys.filter((key) => {
    if (key.kty !== 'RSA') {
        return false;
    }
    if (!key.kid) {
        return false;
    }
    if (key.hasOwnProperty('use') && key.use !== 'sig') {
        return false;
    }
    return ((key.x5c && key.x5c.length) || (key.n && key.e));
})
```
Your thoughts on this?

File: tests/jwksClient.tests.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (manpreet-compro) [reply tests/jwksClient.tests.js]: Yes that is better. Incorporated

File: src/JwksClient.js (hunk 1 above)
Comment (lbalmaceda) [reply src/JwksClient.js]: Yes, it does. Go ahead 👍

Comment (manpreet-compro) [reply src/JwksClient.js]: Done.

**Pull request merged**